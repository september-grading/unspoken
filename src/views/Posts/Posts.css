.post-list {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  align-items: center;
  align-content: center;
}
