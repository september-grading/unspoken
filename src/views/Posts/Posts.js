import React, { useEffect, useState } from 'react';
import Preview from '../../components/Preview/Preview';
import { getPosts } from '../../services/posts';

export default function Posts() {
  const [posts, setPosts] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      const resp = await getPosts();
      setPosts(resp);
      setLoading(false);
    };
    fetchData();
  }, [loading]);

  if (loading) {
    return <h3>Loading...</h3>;
  }

  return (
    <div>
      {posts.map((item) => (
        <Preview key={item.id} {...item} />
      ))}
    </div>
  );
}
