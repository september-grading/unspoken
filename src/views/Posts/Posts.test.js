import { MemoryRouter } from 'react-router-dom';
import Posts from '../Posts/Posts';
import { render, screen } from '@testing-library/react';

test('should render list of posts', async () => {
  const { container } = render(
    <MemoryRouter>
      <Posts />
    </MemoryRouter>
  );
  await screen.findByText('Tokio and Nori');
  expect(container).toMatchSnapshot();
});
