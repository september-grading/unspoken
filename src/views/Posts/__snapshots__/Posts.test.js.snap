// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list of posts 1`] = `
<div>
  <div
    class="post-list"
  >
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="click-post"
          href="/posts/36"
        >
          <div
            class="title"
          >
            ta overload
          </div>
        </a>
        <div
          class="content"
        >
          we have to remember what we did 5 months ago lol
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="click-post"
          href="/posts/50"
        >
          <div
            class="title"
          >
            we are almost done
          </div>
        </a>
        <div
          class="content"
        >
          almost done LESSSSGOOOOOOOO
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="click-post"
          href="/posts/51"
        >
          <div
            class="title"
          >
            benny
          </div>
        </a>
        <div
          class="content"
        >
          benny is the cutest and most powerfulest dog
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="click-post"
          href="/posts/37"
        >
          <div
            class="title"
          >
            Tokio and Nori
          </div>
        </a>
        <div
          class="content"
        >
          cuties
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="click-post"
          href="/posts/52"
        >
          <div
            class="title"
          >
            writing a post
          </div>
        </a>
        <div
          class="content"
        >
          I need to see the edit and delete buttons to finish coding lol
        </div>
      </div>
    </div>
    <div
      class="preview-card-container"
    >
      <div
        class="preview-card"
      >
        <a
          class="click-post"
          href="/posts/53"
        >
          <div
            class="title"
          >
            Dear Julie...
          </div>
        </a>
        <div
          class="content"
        >
          I'm gonna write something to try and make Julie cry here
        </div>
      </div>
    </div>
  </div>
  <div
    class="page-nav"
  >
    <div
      class="page-number"
    >
      Page: 
      1
    </div>
    <button>
      Previous Page
    </button>
    <button>
      Next Page
    </button>
  </div>
</div>
`;
